faunadb/client_logger.py

```diff
@@ -0,0 +1,46 @@
+from ._json import to_json
+
+
+# Function suitable to be passed
+def logger(logger_func):
+  """
+  Function that can be the ``observer`` for a :any:`Client`.
+  :param logger_func:
+      Callback taking a string to be logged.
+  """
+  return lambda info: logger_func(show_request(info))
+
+
+def show_request(info):
+  """Translates a :any:`RequestResponseInfo` to a string suitable for logging."""
+  rq = info.request
+  rs = info.response
+  parts = []
+  log = parts.append
+
+  log("Fauna %s /%s%s\n" % (rq.method, rq.path, _query_string(rq.query)))
+  auth = info.client.session.auth
```

Comment:
Two comments:

1. I think we can more this line up to where we are grabbing out the request and response
2. Maybe we should provide a helper in `RequestResponseInfo` to access auth for us, so you don't have to go through the session to get the auth.